opencti-platform/opencti-graphql/src/graphql/authDirective.js

```diff
@@ -76,18 +76,26 @@ export const authDirectiveBuilder = (directiveName) => {
                 }
                 return null;
               }
-              // Check the user capabilities
-              let numberOfAvailableCapabilities = 0;
-              for (let index = 0; index < requiredCapabilities.length; index += 1) {
-                const checkCapability = requiredCapabilities[index];
-                const matchingCapabilities = filter((r) => checkCapability !== BYPASS && includes(checkCapability, r), userCapabilities);
-                if (matchingCapabilities.length > 0) {
-                  numberOfAvailableCapabilities += 1;
-                }
-              }
-              const isAccessForbidden = numberOfAvailableCapabilities === 0
-                || (requiredAll && numberOfAvailableCapabilities !== requiredCapabilities.length);
-              if (isAccessForbidden) {
+              let userCapabilities = [];
+
+              const isInDraftContext = !!getDraftContext(context, user);
+              // If the user is in draft mode, add capabilities in draft to the base capabilities 
+              if (isInDraftContext) {
+                const userCapabilitiesInDraft = map((c) => c.name, user.capabilitiesInDraft);
+                userCapabilities = Array.from(new Set([...userBaseCapabilities, ...userCapabilitiesInDraft]));
```

Comment:
we could keep the Set since we are looking for element in it line 92

Reply:
We'll stick with the array as now we use `.some` on it first